### PR TITLE
LPD-36743 Add ExternalReferenceCode to Categories REST API's

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-35700 Add ExternalReferenceCode to Categories REST API's

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-36743

ERC’s for Categories/Vocabularies are already applied at the service level, but some of the REST API’s do not expose them. The purpose of this story is to add ERC to schemas which are missing them.

# How am I fixing it?

I'm adding the erc on the yaml file for the parentTaxonomyCategory, parentTaxonomyVocabulary, and TaxonomyCategoryProperty

- TaxonomyCategoryProperty > Object AssetCategoryProperty do not have the erc so I’m adding it
- PageTaxonomyCategory and PageTaxonomyVocabulary are “pages” of TaxonomyCategory and TaxonomyVocabilary so the ERC is inside each element now.
- TaxonomyCategoryBrief > we will use the ERC on taxonomyCategoryReference

# How can you verify that it works?

Run the included automated tests.

# Commits

- **LPD-36743 add ERC to AssetCategoryProperty**
- **LPD-36743 buildService**
- **LPD-36743 Add upgrade process for ERC to AssetCategoryProperty table**
- **LPD-36743 TaxonomyCategoryDTOConverter.java**
- **LPD-36743 Add test for upgrade process**
- **LPD-36743 implement the getCategories on the AssetEntryAssetCategoryRelAssetCategoryLocalServiceWrapper to obtain the data for the TaxonomyCategoryBriefs**
- **LPD-36743 fix call for wiki dto**
- **LPD-36743 WIP remove FF**
- **LPD-36743 add externalReferenceCode to parentTaxonomyCategory, TaxonomyCategoryProperty, and parentTaxonomyVocabulary**
- **LPD-36743 buildRest**
- **LPD-36743 add external reference code to objects dto converter**
- **LPD-36743 check on test that the value of the erc**
- **LPD-36743 add test to check that once updated the properties of a category, the category show the field with erc**
- **LPD-36743 check the values of the erc on the parentTaxonomyCategory on tests**
- **LPD-36743 add test to check that on BlogPosting if there is categories the TaxonomyCategoryBrief is correct**
- **LPD-36743 add some test to WikiPage to assert that now the TaxonomyCategoryBrief is returned with the category**

### CODE REVIEW

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-test/src/testIntegration/java/com/liferay/headless/admin/taxonomy/resource/v1_0/test/TaxonomyCategoryResourceTest.java [hunk 1]
```diff
@@ -233,6 +235,43 @@ public void testGetTaxonomyCategoryTaxonomyCategoriesPageWithSortDateTime()
...
+			taxonomyCategoryProperty.toString(),
+			taxonomyCategoryProperty.getExternalReferenceCode());
+		Assert.assertEquals(
+			taxonomyCategoryProperty.toString(),
+			taxonomyCategoryProperty.getKey(), key);
+		Assert.assertEquals(
+			taxonomyCategoryProperty.toString(),
+			taxonomyCategoryProperty.getValue(), value);
+	}
+
```
[the remark this answers is not in the document]
Comment (AliciaGarciaGarcia) [reply modules/apps/headless/headless-admin-taxonomy/headless-admin-taxonomy-test/src/testIntegration/java/com/liferay/headless/admin/taxonomy/resource/v1_0/test/TaxonomyCategoryResourceTest.java:274]: thanks!

I will look where should belong to :)

**Comment by AliciaGarciaGarcia** [comment]
add the suggested changes and rebased with master

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
resolve conflicts and rebase with master

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by adolfopa** [comment]
ci:stop

**Comment by adolfopa** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
my bad. 

I must have accidentally deleted something on the rebase.

I'll rebase and resend.

**Comment by AliciaGarciaGarcia** [comment]
the other stable failure is common and from obs :(

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Pull request closed without merging**